ui/tests/integration/components/filter-input-explicit-test.js

```diff
@@ -0,0 +1,66 @@
+/**
+ * Copyright (c) HashiCorp, Inc.
+ * SPDX-License-Identifier: BUSL-1.1
+ */
+
+import { module, test } from 'qunit';
+import { setupRenderingTest } from 'ember-qunit';
+import { render, typeIn, click } from '@ember/test-helpers';
+import hbs from 'htmlbars-inline-precompile';
+import { GENERAL } from 'vault/tests/helpers/general-selectors';
+import sinon from 'sinon';
+
+module('Integration | Component | filter-input-explicit', function (hooks) {
+  setupRenderingTest(hooks);
+
+  hooks.beforeEach(function () {
+    this.handleSearch = sinon.spy();
+    this.handleInput = sinon.spy();
+    this.handleKeyDown = sinon.spy();
+    this.query = '';
+    this.placeholder = 'Filter roles';
+
+    this.renderComponent = () => {
+      return render(
+        hbs`<FilterInputExplicit aria-label="test-component" @placeholder={{this.placeholder}} @query={{this.query}} @handleSearch={{this.handleSearch}} @handleInput={{this.handleInput}} @handleKeyDown={{this.handleKeyDown}} />`
+      );
+    };
+  });
+
+  test('it renders', async function (assert) {
+    assert.expect(2);
+    this.query = 'foo';
+    await this.renderComponent();
+
+    assert
+      .dom(GENERAL.filterInputExplicit)
+      .hasAttribute('placeholder', 'Filter roles', 'Placeholder passed to input element');
+    assert.dom(GENERAL.filterInputExplicit).hasValue('foo', 'Value passed to input element');
+  });
+
+  test('it should call handleSearch on submit', async function (assert) {
+    assert.expect(1);
+
+    this.handleSearch = () => {
+      assert.ok(true, 'handleSearch was called');
+    };
+
+    await this.renderComponent();
+    await typeIn(GENERAL.filterInputExplicit, 'bar');
+    await click(GENERAL.filterInputExplicitSearch);
+  });
+
+  test('it should send keydown event on keydown', async function (assert) {
+    assert.expect(3); // handle key down called twice
+
+    this.handleKeyDown = () => {
+      assert.ok(true, 'handleKeydown was called');
+    };
+
+    await this.renderComponent();
+    await typeIn(GENERAL.filterInputExplicit, 'a');
+    await typeIn(GENERAL.filterInputExplicit, 'b');
+
+    assert.ok(this.handleSearch.notCalled);
```

Comment:
```suggestion
    assert.true(this.handleSearch.notCalled);
```
or `assert.false(this.handleSearch.called)` (take your pick)

pro tip - i'd strongly advise against using `assert.ok` or `assert.notOk` because they pass on truthy values instead of strict equality. you can read more about it [here](https://github.com/platinumazure/eslint-plugin-qunit/blob/main/docs/rules/no-assert-ok.md).

ideally we should enforce this rule in ESLint but i keep forgetting to bring it up. 😛

Reply:
Learning a lot, thank you for the comments and links